site/cds_rdm/legacy/redirector.py

```diff
@@ -59,19 +55,62 @@ def legacy_files_redirect(legacy_id, filename):
     return redirect(url_path)
 
 
+def legacy_collection_redirect(collection_name):
+    """Redirection for legacy collections."""
+    cds_community_uuid = current_app.config["CDS_REDIRECTION_COLLECTIONS_MAPPING"].get(
+        collection_name, None
+    )
+    if not cds_community_uuid:
+        raise NoResultFound
+    url_path = url_for(
+        "invenio_app_rdm_communities.communities_home",
+        pid_value=cds_community_uuid,
+        **request.args,
+    )
+    return redirect(url_path)
+
+
+def legacy_search_redirect():
+    """
+    Redirection for legacy search. Transforms the legacy URL syntax into RDM URL syntax.
+
+        /legacy?cc=<legacy collection name>... -> /communities/<rdm_community_id>/records?...
+        /legacy?c=<legacy collection name>... -> /communities/<rdm_community_id>/records?...
+        /legacy?c=<legacy collection name>&p=<query>... -> /communities/<rdm_community_id>/records?q=<query>...
+    """
+    # Fetch current collection if it exists
+    collection_name = request.args.get("cc", None)
+    # If not, then fetch from collection list, URLs with only single 'c' will be redirected for now
+    if not collection_name:
+        collection_name = request.args.get("c", None)
```

Comment:
`c` can be a list, are we getting the first?

Reply:
YEs, we will only be redirecting URLs with a single c. I am setting this condition in the legacy apache config, so if there are multiple 'c' params, it won't get redirected here anyways.